### PR TITLE
added missing dependencies causing the ambiguous 'Build plan failed.'…

### PR DESCRIPTION
**Pull request by ekalosak**
… error raised when running the ./Shakefile.hs previously; Updated the README.md with a beginner-friendly 'getting started' section.

### CODE REVIEW

**Comment by ekalosak** [comment]
@mfine Shake and Lint are required to run the `./Shakefile.hs`, building on a clean Debian 4.9.0 VirtualBox. Without Shake, an ambiguous "Build plan failed" error surfaces. Without Lint, the `./Shakefile.hs lint` command fails.

Understanding the `stack.yaml` file may not be the most appropriate place to note these dependencies, where's a better place?

**Comment by mfine** [comment]
If you need to install them and specify them, you can do that outside of the project in the global stack.yaml `~/.stack/global-project/stack.yaml` as they're not formally part of the project and are used as globally present tooling, not proper project dependencies.

**Comment by mfine** [comment]
Another way we've gone about expressing dependencies like this is with nix files. See https://github.com/swift-nav/hitl/blob/master/default.nix.

**Comment by ekalosak** [comment]
Problem is, without `shake` in the `stack.yaml` `extra-deps` section, the stack build fails on travis. That is, without `shake-0.16.4` as an explicit requirement, the ugly `Plan construction failed.` error rears its head.

As an aside, the docker build works just fine - I think that's because it builds from an intermediate on the private ECR that has `shake` all shook out correctly. But for clean installs on arbitrary hosts, the `shake` dependency is stiff. And it's a bit self evident - we need `shake` to run `./Shakefile.hs`.

**tl;dr** We really aught to have `shake-0.16.4` in `stack.yaml`.

**Comment by mfine** [comment]
What about then installing shake explicitly in the .travis.yml instead?

```yaml
install:
  - travis_retry curl -L https://www.stackage.org/stack/linux-x86_64 | tar xz --wildcards --strip-components=1 -C ~/.local/bin '*/stack'
  - stack config set system-ghc --global true
  - stack install shake shake-0.16.3
  - pip install --user --upgrade awscli
```

Using the shake version from the resolver https://www.stackage.org/lts-10.10/package/shake-0.16.3.

**Comment by ekalosak** [comment]
Asking Travis to install `shake` outside of `stack` via the `pip install` suggested above still causes the [breakage](https://travis-ci.org/swift-nav/wolf/builds/439784517). Either `stack` or `shake` appear not to like working with the global haskell environment here. Suggestions?

**Comment by mfine** [comment]
Hmmm, weird - not sure what's going on. Tried: https://github.com/swift-nav/wolf/pull/48/files - but that didn't work either. Maybe put the shakers dependency in the extra-deps instead of shake directly and call it a day?

**Comment by mfine** [comment]
Here was the commit where we stopped using shakers in the extra deps section:

https://github.com/swift-nav/wolf/commit/5bf19ef36e9ad8a35b7d00f033ccbb486afd1e98#diff-fafd0cdcd559a7b124cc61c29413fb54

**Comment by ekalosak** [comment]
Build is passing, @mfine should I merge?

**Comment by mfine** [comment]
Let's merge https://github.com/swift-nav/wolf/pull/49 and close this.

**Comment by mfine** [comment]
#49 made some tweaks on this. Pulled that in. Thanks!
